dot/peerset/handler.go

```diff
@@ -113,40 +115,29 @@ func (h *Handler) PeerReputation(peerID peer.ID) (Reputation, error) {
 	if err != nil {
 		return 0, err
 	}
-	return n.getReputation(), nil
+	return n.reputation, nil
+}
+
+// SetMessageProcessor set a processor to process peerset messages
+func (h *Handler) SetMessageProcessor(processor MessageProcessor) {
+	h.peerSet.processor = processor
 }
 
 // Start starts peerSet processing
 func (h *Handler) Start(ctx context.Context) {
-	ctx, cancel := context.WithCancel(ctx)
-	h.cancelCtx = cancel
-
-	actionCh := make(chan action, msgChanSize)
-	h.closeCh = make(chan struct{})
-	h.actionQueue = actionCh
-
-	h.peerSet.start(ctx, actionCh)
+	h.peerSet.start(ctx)
 }
 
-// SortedPeers return chan for sorted connected peer in the peerSet.
-func (h *Handler) SortedPeers(setIdx int) chan peer.IDSlice {
-	resultPeersCh := make(chan peer.IDSlice)
-	h.actionQueue <- action{
-		actionCall:    sortedPeers,
-		resultPeersCh: resultPeersCh,
-		setID:         setIdx,
-	}
-
-	return resultPeersCh
+// SortedPeers returns a sorted peer ID slice for connected peers in the peerSet.
+func (h *Handler) SortedPeers(setIdx int) peer.IDSlice {
+	return h.peerSet.peerState.sortedPeers(setIdx)
 }
 
-// Stop closes the actionQueue and result message chan.
-func (h *Handler) Stop() {
-	select {
-	case <-h.closeCh:
-	default:
-		h.cancelCtx()
-		close(h.closeCh)
-		close(h.actionQueue)
+func stringfyPeers(peers peer.IDSlice) string {
```

Comment:
nit should we have this as a `String()` method on `peer.IDSlice` instead?

Reply:
the `peer.IDSlice` is from `libp2p` external package

Reply:
Btw, I found useful to make a PR to add this -> https://github.com/libp2p/go-libp2p-core/pull/238